### ISSUE TITLE
Flutter running on iOS emulator experiences severe lag

### ISSUE THREAD

**Issue by Deathislikethewind** [flutter/flutter#143738]
### Steps to reproduce

Demo

```dart
class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;
  late TextEditingController _controller;
  @override
  void initState() {
    // TODO: implement initState
    super.initState();
    _controller = TextEditingController(text: '');
  }

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const Text(
              'You have pushed the button this many times:',
            ),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.headlineMedium,
            ),
            const TextField(),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}
```

Running on the iOS simulator, any model can handle this situation, and it will occur

### Actual results

When I run on an iOS simulator, as long as the code contains textfield and I input some content, the simulator will start to lag, accompanied by the entire computer's lag. However, once I kill the app or switch to a page without textfield, the lag will be immediately lifted and everything will return to normal. Is this issue related to the Flutter code? Because I looked at my memory and CPU, and in both running and stopping the app, I saw approximate space and utilization ratios, both remaining. The total memory size was 32GB, with 25GB used, while the CPU was about 80% idle. And I initially thought it was a bug in Xcode, so I tried creating a new project using Xcode and adding textfield. Unfortunately, the app created by Xcode did not experience any lag. And I have noticed that when I use Flutter for development, sometimes I encounter situations where I cannot hot reload and prompt the following message: "Hot reload was rejected:"

Error while starting Kernel isolate task

Try performing a hot restart instead “
I don't know if it's related to lag, I don't know how to handle it, and I tried using Xcode to open the project instead of Flutter Run to run the code, but it still caused lag in the simulator and computer

### Logs

<details open>
<summary>Logs</summary>

```console

none

```

</details>

### Flutter Doctor output

<details open>
<summary>Doctor output</summary>

```console

➜  ~ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.10.0, on macOS 14.2.1 23C71 darwin-x64, locale
    zh-Hans-CN)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.3)
[✓] VS Code (version 1.86.0)
[✓] Connected device (5 available)
[✓] Network resources


```

</details>

**Comment by huycozy** [comment]
Hi @Deathislikethewind 

>And I have noticed that when I use Flutter for development, sometimes I encounter situations where I cannot hot reload and prompt the following message: "Hot reload was rejected:"
>Error while starting Kernel isolate task
>Try performing a hot restart instead “

When the issue occurs (app and host machine become laggy), these logs always appear in the output console, right? If so, could you share the entire output log (`flutter run -v`)? If the log is too long, please save it into a file and attach it here.

Also, I see you are using an old Flutter version, please try upgrading to the latest version and check again to see if the issue persists or not, see [Upgrading Flutter](https://docs.flutter.dev/release/upgrade).

**Comment by Deathislikethewind** [comment]
There is no log output for this issue

I tried upgrading the Flutter version to 3.19.0 and found that the input text lag issue in the demo project was resolved. However, when I ran the old project, I encountered a problem that I couldn't handle. I had to roll back the Flutter version to run my iOS project, while Android was not affected. After updating the Flutter version, I encountered the following issue

Error (Xcode): underlying Objective-C module 'IosAwnCore' not found
/Users/xxx/Library/Developer/Xcode/DerivedData/Runner-aquoqxhtpsttqogqkfdsghmvgrcp/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/IosAwnCore.build/Objects-normal/x86_64/IosAwnCore.swiftinterface:6:18

Error (Xcode): failed to verify module interface of 'IosAwnCore' due to the errors above; the textual interface may be broken by project issues or a compiler bug
/Users/xxx/Library/Developer/Xcode/DerivedData/Runner-aquoqxhtpsttqogqkfdsghmvgrcp/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/IosAwnCore.build/Objects-normal/x86_64/IosAwnCore.swiftinterface:0:0

Error (Xcode): underlying Objective-C module 'IosAwnCore' not found
/Users/xxx/Library/Developer/Xcode/DerivedData/Runner-aquoqxhtpsttqogqkfdsghmvgrcp/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/IosAwnCore.build/Objects-normal/x86_64/IosAwnCore.private.swiftinterface:6:18

Error (Xcode): underlying Objective-C module 'IosAwnCore' not found
/Users/xxx/Library/Developer/Xcode/DerivedData/Runner-aquoqxhtpsttqogqkfdsghmvgrcp/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/IosAwnCore.build/Objects-normal/arm64/IosAwnCore.swiftinterface:6:18

flutter clean, restart the computer, restart Xcode but none of these operations solved the above problem
